akka-actor/src/main/resources/reference.conf

```diff
@@ -351,16 +351,61 @@ akka {
       }
 
       affinity-pool-executor {
-        # The number of threads that this executor will maintain in its pool
-        num-threads = 8
+        # Min number of threads to cap factor-based parallelism number to
+        parallelism-min = 8
+
+        # The parallelism factor is used to determine thread pool size using the
+        # following formula: ceil(available processors * factor). Resulting size
+        # is then bounded by the parallelism-min and parallelism-max values.
+        parallelism-factor = 3.0
+
+        # Max number of threads to cap factor-based parallelism number to
+        parallelism-max = 64
 
-        # The max size of the queue that is used by each thread in the pool
-        # If the queue of a particular worker is full, no more tasks can be
-        # enqueued until space frees up
+        # Each worker in the pool uses a seprate bounded MPSC queue. This value
+        # indicates the upper bound of the queue. Whenever an attempt to enqueue
+        # a task is made and the queue does not have capacity to accomodate
+        # the task, RejectedExecutionException is thrown.
         affinity-group-size = 10
 
-        # A list of the affinity CPU strategies that the pool will useU
+        # A list of affinity CPU strategies, ordered by preference. The thread factory
+        # backing up this executor shall use the strategies to try and construct a thread
+        # that is on the same/different core/socket. Allowed values are:
+        #
+        # - any - no affinity
+        # - same-core - must be a CPU on the same core
+        # - same-socket - must be on the same socket but on different core
+        # - different-core - must be a cpu on any other core
+        # - different-socket - must be a CPU on any other socket
+        #
+        # It is important to note that in any case "any" should be present in
+        # the list of strategies as the last one in terms of preference. as
+        # explained in:
+        # https://github.com/OpenHFT/Java-Thread-Affinity/blob/master/affinity/src/main/java/net/openhft/affinity/AffinityLock.java#L258
+        #
+        # For more info on affinity: https://github.com/peter-lawrey/Java-Thread-Affinity/
+
         cpu-affinity-strategies = [any]
+
+        # The thread pool worker spins in a loop, repeatedly polling
+        # its queue to obtain a task and execute it. When it fails to do
+        # so, it continues to loop. The waiting strategy determines the behavior
```

Comment:
The sentence "When it fails to do so, it continues to loop." is superfluous, delete. :)